### PR TITLE
MAT-3865 Save test case as embedded document in measure, move testc…

### PR DESCRIPTION
**Pull request by adongare**
MAT-3865 Save test case as embedded document in measure, move test cases to measure service

## MADiE PR

Jira Ticket: [MAT-3865](https://jira.cms.gov/browse/MAT-3865)
(Optional) Related Tickets:

### Summary
- Moved test cases to measure-service as an embedded document in measure document
- Did not add lazy loading

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Review by jkotanchik-SB — DISMISSED** [review]

File: src/main/java/cms/gov/madie/measure/services/TestCaseService.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  public List<TestCase> findTestCasesByMeasureId(String measureId) {
+    return findMeasureById(measureId).getTestCases();
+  }
+
+  public Measure findMeasureById(String measureId) {
+    Measure measure = measureRepository.findById(measureId).orElse(null);
+    if (measure == null) {
+      throw new ResourceNotFoundException("Measure", measureId);
```
Comment (nmorasb) [inline src/main/java/cms/gov/madie/measure/services/TestCaseService.java:42]: I think the way this is written right now would return a 500 to the caller, which probably isn't what we want. In the past I have done a few different things with this kind of check:
- handle this check at the controller level, and throw a ResponseStatusException (provided by Spring). This is very configurable per instance and the error messages returned would look just like any other errors caught by Spring.
- Use @ControllerAdvice to create a global exception handler and map different internal application errors to different response codes / objects
- Handle this check at the controller (or catch this exception) and return a ResponseEntity.notFound() (or similar)

Unless calling the API with an unknown measure ID is truly an internal exception, I think we should handle that and return some nicer response code and message.

Comment (adongare) [reply src/main/java/cms/gov/madie/measure/services/TestCaseService.java:42]: I thought we have ControllerAdvice that catches the exception and prepares the error response. I'll double-check with frontend changes once I pull your security branch.

Comment (adongare) [reply src/main/java/cms/gov/madie/measure/services/TestCaseService.java:42]: @nmorasb, there was no exception handler. I've added an exception handler in controller advice with BAD_REQUEST status code. pls, let me know if this works

Comment (jkotanchik-SB) [reply src/main/java/cms/gov/madie/measure/services/TestCaseService.java:42]: @adongare How about http status NOT_FOUND instead? It informs the caller that their request is acceptable, but the requested resource isn't available at the path.

Comment (adongare) [reply src/main/java/cms/gov/madie/measure/services/TestCaseService.java:42]: @jkotanchik-SB can do that. was not sure if NOT_FOUND or BAD_REQUEST. Thanks!

**Review by jkotanchik-SB — DISMISSED** [review]

File: src/main/java/cms/gov/madie/measure/services/TestCaseService.java (hunk 1 above)
Comment (adongare) [reply src/main/java/cms/gov/madie/measure/services/TestCaseService.java:42]: done

**Review by jkotanchik-SB — APPROVED** [review]
